axum/src/routing/route.rs

```diff
@@ -155,9 +155,6 @@ where
 
     #[inline]
     fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
-        #[derive(Clone, Copy)]
-        struct AlreadyPassedThroughRouteFuture;
```

Comment:
Maybe dig up the PR that introduced this via git blame?

Reply:
I did that [here](https://github.com/tokio-rs/axum/pull/1711/files#r1082406463)

> tbh I'm not totally sure why everything still worked after removing this. Doesn't seem related to this change because it also works on `main` if you remove this. So must be something else that changed 🤷 
> 
> https://github.com/tokio-rs/axum/pull/755 did add a regression test and I also tried the repro from https://github.com/tokio-rs/axum/issues/747. Both worked correctly.

and tested specifically the repro from the issue and everything worked fine. So something else must have changed that made `AlreadyPassedThroughRouteFuture` unnecessary.

Reply:
Ah, great 🙂